multiload/linux-proc.c

```diff
@@ -252,24 +252,24 @@ GetPage (int Maximum, int data [3], LoadGraph *g)
 void
 GetMemory (int Maximum, int data [5], LoadGraph *g)
 {
-    int user, shared, buffer, cached;
+    int user, shared, cached, free;
 
     glibtop_mem mem;
 
     glibtop_get_mem (&mem);
 
     g_return_if_fail ((mem.flags & needed_mem_flags) == needed_mem_flags);
 
-    user    = rint (Maximum * (float)mem.user / (float)mem.total);
+    user    = rint (Maximum * ((float)mem.user - (float)mem.shared) / (float)mem.total);
     shared  = rint (Maximum * (float)mem.shared / (float)mem.total);
-    buffer  = rint (Maximum * (float)mem.buffer / (float)mem.total);
-    cached = rint (Maximum * (float)mem.cached / (float)mem.total);
+    cached  = rint (Maximum * ((float)mem.total - (float)mem.user - (float)mem.free) / (float)mem.total);
+    free    = rint (Maximum * (float)mem.free / (float)mem.total);
 
     data [0] = user;
     data [1] = shared;
-    data [2] = buffer;
+    data [2] = 0;
```

Comment:
Why are you doing this, ie is it a data source or not?

Reply:
It has to be passed to keep the colors from "sliding".

Reply:
Please, check how the graph is made in order to avoid this memory waste.

Reply:
You only use/need 3 data sources. (user, shared, cached).
free (background) = 100 - (user + shared + cached)

Reply:
The graph is made in such a way that it takes all colors from gsettings, and expects them to be named sequentially. Therefore we have two possible solutions:
- Rename the colors, and accept a little breakage in terms of the upgrade process
- Use this "dummy color" and waste four bytes of memory to ensure effortless upgrade

I could not find exactly where the graph is made (this is literally the first thing I'm writing in C, and I have no idea about GTK), but I did understand *how* it seems to be made.

Reply:
https://github.com/mate-desktop/mate-applets/blob/01e8f8a09ab273719bdf0869f9f8693cb45d302e/multiload/load-graph.c#L47-L49